src/librustc_typeck/check/mod.rs

```diff
@@ -512,6 +512,19 @@ impl<'tcx> EnclosingBreakables<'tcx> {
     }
 }
 
+/// This is returned by `get_node_fn_decl` to decide whether to suggest
+/// a return type
+pub enum FnDeclType {
```

Comment:
I'm wondering whether adding two fields with `&'tcx hir::FnDecl` and `ast::Ident` to these variants and adding another variant for the `Option::None` case would make the code easier to understand or not (mainly because of the changes that would end up in `get_fn_decl` and its callees).

Reply:
I'd like to have everything working as intended before I refactor the code, but yes, I agree that putting more information in the type will make the rest of the code easier to read.